scenarios/aws/eks/run.go

```diff
@@ -271,24 +272,25 @@ func Run(ctx *pulumi.Context) error {
 		copy(workloadWithCRDDeps, workloadDeps)
 
 		if awsEnv.AgentDeploy() {
-			helmComponent, err := agent.NewHelmInstallation(&awsEnv, agent.HelmInstallationArgs{
-				KubeProvider:  eksKubeProvider,
-				Namespace:     "datadog",
-				Fakeintake:    fakeIntake,
-				DeployWindows: awsEnv.EKSWindowsNodeGroup(),
-			}, utils.PulumiDependsOn(workloadDeps...))
-			if err != nil {
-				return err
-			}
+			k8sAgentOptions := make([]kubernetesagentparams.Option, 0)
+			k8sAgentOptions = append(
+				k8sAgentOptions,
+				kubernetesagentparams.WithNamespace("datadog"),
+				kubernetesagentparams.WithFakeintake(fakeIntake),
+				kubernetesagentparams.WithPulumiResourceOptions(utils.PulumiDependsOn(workloadDeps...)),
+			)
 
-			ctx.Export("agent-linux-helm-install-name", helmComponent.LinuxHelmReleaseName)
```

Comment:
I think removing the export match what was advised [here](https://github.com/DataDog/test-infra-definitions/pull/836#issuecomment-2127602752). Because the `KubernetesAgent` component should not have Helm specific fields.
However I think you are using that export in your e2e tests @vboulineau . What do you think would be the way to change that so it does not break your tests?

Reply:
From what I can see, we need it to build label selectors. I think we could export like:
```
KubernetesObjectRef:
  namespace:
  name:
  kind:
  apiVersion:
  labelSelectors:

KubernetesAgent:
  nodeAgent: <KubernetesObjectRef>
  clusterAgent: <KubernetesObjectRef>
  clusterCheckRunner: <KubernetesObjectRef>

Reply:
Thanks @vboulineau! I've added a new `KubernetesObjectRef` component here: https://github.com/DataDog/test-infra-definitions/pull/881/files#diff-fd090de4fed44a578bc1e3d8e1af4d8518fdf506dd390f052607e8bf4de995f4R23-R33

I set `version` and `appVersion` with the helmComponent values. Not sure what `kind` should refer to, but at the moment, I set them to `Pod`. Let me know if I should make any updates to the new values